src/ScottPlot5/ScottPlot/Range.cs

```diff
@@ -30,6 +30,11 @@ public Range(double min, double max)
                 throw new ArgumentException($"Argument ${nameof(max)} must be finite and non-NaN");
             }
 
+            if (double.IsNaN(min) || double.IsNaN(max))
```

Comment:
This check already exists above.

FWIW, there's a `double.IsFinite` which ensures it's non-NaN and not ±infinity, but we can't use it on all the things we target. Since we use this logic a lot it might make sense to create our own.